lib/pbench/test/unit/server/conftest.py

```diff
@@ -403,17 +403,17 @@ def fake_find(name: str) -> Template:
                 },
                 version=5,
             )
-        elif name == "result":
+        elif name == "result-data-sample":
             return Template(
-                name="result",
-                idxname="result-data",
-                template_name="unit-test.v5.result-data",
+                name="result-data-sample",
+                idxname="result-data-sample",
+                template_name="unit-test.v6.result-data-sample",
                 file="run.json",
-                template_pattern="unit-test.v5.result-data.*",
-                index_template="unit-test.v5.result-data.{year}-{month}",
+                template_pattern="unit-test.v6.result-data.*",
+                index_template="unit-test.v6.result-data-sample.{year}-{month}",
                 settings={"none": False},
                 mappings={
-                    "_meta": {"version": "5"},
+                    "_meta": {"version": "6"},
```

Comment:
No its just to make the unit tests simpler by keeping the version same for all the documents, so we dont have to mange the version numbers across multiple api tests. Like run-data documents are version 6 but result-data-sample documents are version 5, so if we keep them different we have to make sure we get the right version number when dynamically building the indices. Keeping them same makes it simpler

Reply:
OK, I'm just not aware of what the version numbers really mean, what they are used for, and by whom, so I don't understand what the implications of changing them are.  I realize that they have to be set to _something_, and I concur that it would be best if they were set in a realistic fashion (e.g., so that the documents match the index they are supposedly coming from, etc.), but I also think that they should be meaningful, that they should serve a purpose (i.e., that we're not maintaining them for no reason), and that, therefore, they only change when they are supposed to.

Reply:
Hmm; this is fine to some extent, however the various index templates *do* have different versions, and it was somewhat nice to be testing that this carries through to the application code. I'm not sure how I feel about this... but it bothers me just enough to unresolve Webb's conversation and make everyone think about it again.